config/routes.rb

```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get 'users', to: 'user#index', as: 'users'
+  get '/user/:id', to: 'user#show', as: 'user'
+  get '/post/:user_id/posts', to: 'post#index', as: 'user_posts'
+  get '/post/:user_id/posts/:id', to: 'post#show', as: 'user_post'
```

Comment:
- [ ] I suggest that you use `resources` because Resource routing allows you to quickly declare all of the common routes for a given resourceful controller
```
 resources: users, only: [: index, : show] do
    resources: posts, only: [: index, : show]
  end
```